### PR TITLE
Don't show feedback snackbar till marketplace content has finished loading

### PR DESCRIPTION
**Pull request by andfinally**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes 17942-gh-Automattic/woocommerce.com.

- Ensures we only show the snackbar inviting feedback some time after the content has loaded.
  - We set a date `marketplace_redesign_2023_last_shown_date` in local storage to ensure we only show one snackbar.
- Ensures snackbar inherits fixed position from `.woocommerce-layout__footer`, so it's visible wherever you are on the page.
  - Removes `WooFooterItem` from around marketplace `Footer`, so it's no longer a child of the WooCommerce Admin footer.
  - Removes the `position: relative` from `.woocommerce-layout__footer`.
- Changes use of context. Context for marketplace now only has states for the selected tab and loading state.
  - Renames `ProductListContext`, `ProductListContextProvider` and `productListContextValue` to more generic `MarketplaceContext`, `MarketplaceContextProvider` and `marketplaceContextValue`.
  - We use this context in `Tabs` and `Content` to keep track of which tab is selected, and set the selected tab.
  - We also use it in `Discover` and `Extensions`, which both report loading state to the context. This allows us to only render the snackbar when loading is complete.
  - Extensions: moves `productList` and `setProductList` and logic for getting product list out of the context provider to this component. We don't need to share the list of products in the context.
- Renames hostname constant and creates constants for API paths.
 

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. If you have a local storage value `marketplace_redesign_2023_last_shown_date` for your local WooCommerce environment, delete it. (It gets set whenever we show the snackbar, so delete it every time you want to test the snackbar again.)
2. Build WooCommerce Admin on your local environment e.g.: `cd plugins/woocommerce-admin` then `nvm use && pnpm start`.
3. Visit [WooCommerce > Marketplace > Extensions](http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions&tab=extensions).
4. Note that the snackbar with text "How easy is it to find an extension?" only appears about 7 seconds after the content loads on this screen.
5. Click on the `Browse` tab to go to the extensions screen.
6. Note that content for this screen loads, and the snackbar persists, and no second snackbar appears.
7. Delete the local storage value again and reload the page.
8. Note that the snackbar appears about 7 seconds after the content has loaded.
9. Note that the snackbar is positioned fixed relative to the viewport, so it's visible without scrolling.
10. Click around the tabs of the marketplace a bit more to satisfy yourself they're still working OK. Check that the footer looks like it does in trunk.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

Fixed rendering of snackbar inviting feedback on new style marketplace.

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/client/marketplace/components/extensions/extensions.tsx [hunk 1]
```diff
@@ -1,23 +1,87 @@
...
+
+				setProductList( products );
+			} )
+			.catch( () => {
+				setProductList( [] );
+			} )
+			.finally( () => {
+				setIsLoading( false );
+			} );
+	}, [ query ] );
```
Comment (andfinally) [inline plugins/woocommerce-admin/client/marketplace/components/extensions/extensions.tsx]: This logic is moved here from `ProductListContextProvider` (now `MarketplaceContextProvider`). We don't need to keep this component's list of products in context. This matches more closely what we're doing in `Discover`.

**Review by kdevnel — APPROVED** [review]
This is a really nice refactor! I see now why it was a little bit of a wrestle. Thanks for tidying up the context and keeping Discover and Extensions similar - this sets us up well for Phase 2 with themes.

File: plugins/woocommerce-admin/client/marketplace/components/extensions/extensions.tsx (hunk 1 above)
Comment (raicem) [reply plugins/woocommerce-admin/client/marketplace/components/extensions/extensions.tsx]: Oh nice! The less stuff in context, the better for now IMHO.

File: plugins/woocommerce-admin/client/marketplace/components/header-account/header-account.tsx [hunk 2]
```diff
@@ -28,7 +28,7 @@ export default function HeaderAccount(): JSX.Element {
 	// component. That component is either an anchor with href if provided or a button that won't accept an href if no href is provided.
 	// Due to early erroring of TypeScript, it only takes the button version into account which doesn't accept href.
 	// eslint-disable-next-line @typescript-eslint/no-explicit-any
-	const accountURL: any = MARKETPLACE_URL + '/my-dashboard/';
+	const accountURL: any = MARKETPLACE_API_HOST + '/my-dashboard/';
```
Comment (raicem) [inline plugins/woocommerce-admin/client/marketplace/components/header-account/header-account.tsx]: Nitpick: maybe we should call this variable `MARKETPLACE_HOST`. 

`MARKETPLACE_API_HOST + '/my-dashboard/'` looked a little odd to me, considering we're creating a basic webpage URL and not an API endpoint.

**Review by raicem — APPROVED** [review]
These are great changes, done masterfully. Thanks @andfinally.

**Review by Dan-Q — APPROVED** [review]
Seconding @raicem's suggestion; otherwise this looks awesome.

File: plugins/woocommerce-admin/client/marketplace/components/header-account/header-account.tsx (hunk 2 above)
Comment (andfinally) [reply plugins/woocommerce-admin/client/marketplace/components/header-account/header-account.tsx]: Good point! Changed in [`6ae83b0` (#39869)](https://github.com/woocommerce/woocommerce/pull/39869/commits/6ae83b08ecb0d4f500996e50136c509ae1cd94f9).

**Pull request merged**